src/components/WelcomePage/WelcomePageLayout.tsx

```diff
@@ -1,27 +1,35 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 import { WelcomePageLayoutWrapper, TitleStyled, ImageStyled } from './WelcomePageLayout.styles';
 import ArrowDowm from 'assets/arrowDown.jpg';
 import { RootState } from 'redux/store';
-import { useSelector } from 'react-redux';
+import { useAppSelector } from 'redux/hooks';
+import { useNavigate } from 'react-router-dom';
 
 const WelcomePageLayout: FC = () => {
-	const role = useSelector((state: RootState) => state.user.role);
+	const { user } = useAppSelector<RootState>(state => state);
 	const { t } = useTranslation();
+	const navigate = useNavigate();
 
 	const Role = {
 		Freelancer: 'freelancer',
 		Client: 'client',
 	};
 
+	useEffect(() => {
+		if (!user.role) {
+			navigate('/role-selection');
```

Comment:
all path move to const file routes.js or something like that

Reply:
moved